core/domain/learner_group_services.py

```diff
@@ -0,0 +1,647 @@
+    """
+    learner_groups_user_models = (
+        user_models.LearnerGroupsUserModel.get_multi(invited_student_ids))
+
+    for index, student_id in enumerate(invited_student_ids):
+        learner_groups_user_model = learner_groups_user_models[index]
+        if learner_groups_user_model:
+            learner_groups_user_model.invited_to_learner_groups_ids.append(
+                group_id)
+        else:
+            learner_groups_user_model = user_models.LearnerGroupsUserModel(
+                id=student_id,
+                invited_to_learner_groups_ids=[group_id],
+                learner_groups_user_details=[]
+            )
+
+        learner_groups_user_models[index] = learner_groups_user_model
+
+    user_models.LearnerGroupsUserModel.update_timestamps_multi( # type: ignore[type-var]
+        learner_groups_user_models)
+    user_models.LearnerGroupsUserModel.put_multi(learner_groups_user_models) # type: ignore[type-var]
+
+
+def remove_invited_students_from_learner_group(
+    group_id: str,
+    student_ids: List[str]
+) -> None:
+    """Removes the given invited students from the given learner group.
+
+    Args:
+        group_id: str. The id of the learner group.
+        student_ids: list(str). The ids of the students to remove.
+    """
+    found_models = (
+        user_models.LearnerGroupsUserModel.get_multi(student_ids))
+
+    for index, model in enumerate(found_models):
+        # Ruling out the possibility of None for mypy type checking.
+        assert model is not None
+        if group_id in model.invited_to_learner_groups_ids:
+            found_models[index].invited_to_learner_groups_ids.remove(group_id) # type: ignore[union-attr]
+
+    user_models.LearnerGroupsUserModel.update_timestamps_multi(found_models) # type: ignore[type-var]
+    user_models.LearnerGroupsUserModel.put_multi(found_models) # type: ignore[type-var]
+
+
+def get_subtopic_page_progress(
+    user_id: str,
+    subtopic_page_ids: List[str],
+    topics: List[topic_domain.Topic],
+    all_skill_ids: List[str]
+) -> subtopic_page_domain.SubtopicPageSummary:
+    """Returns the progress of the given user on the given subtopic pages.
+
+    Args:
+        user_id: str. The id of the user.
+        subtopic_page_ids: list(str). The ids of the subtopic pages.
+        topics: list(Topic). The topics corresponding to the subtopic pages.
+        all_skill_ids: list(str). The ids of all the skills in the topics.
+
+    Returns:
+        SubtopicPageSummary. Subtopic Page Summary domain object containing
+        details of the subtopic page and users mastery in it.
+    """
+
+    # Fetch the progress of the student in all the subtopics assigned
+    # in the group syllabus.
+    skills_mastery_dict = skill_services.get_multi_user_skill_mastery( # type: ignore[no-untyped-call]
+        user_id, all_skill_ids
+    )
+
+    subtopic_prog_summary: subtopic_page_domain.SubtopicPageSummary
+
+    for topic in topics:
+        assert topic is not None
```

Comment:
Why do you do this? From types we don't expect it to be `None`

Reply:
yup, it's a mistake. Removed this.